### PR TITLE
📁  Feed: Splitting Tests Into Separated Files

### LINKED ISSUE #66: 📁  Feed: Splitting Tests File
In future sight, many more tests will be added.
In order to remain organized the big files must split into small and logically separated files.

Steps to follow:
- [X] Deleted old `tests.py` file
- [X] Add a new `tests` directory inside the feed app root directory
- [X] Add different tests file which will contain the original tests
- [X] Change tests function name's to be more meaningful

### PR DESCRIPTION
**Pull request by paOmer**
This PR aims to take the existing feed app tests and split them into separated files.
In future sight, many more tests will be added, so to remain organized it is best to split big files into small and logically separated files.

- Deleted old `tests.py` file
- Added a new `tests` directory inside the feed app root directory
- Added different tests file which contain the previous tests
- Changed tests function names to be more meaningful

Close #66

### CODE REVIEW

**Review by DeanBiton — APPROVED** [review]
All tests seem to be included.
Good job on improving the names.

**Review by taljacob2 — APPROVED** [review]
LGTM!
Great initiative! 👍

File: feed/tests/test_comment_model.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    comment0 = Comment(author=users[1], post_parent=post0, content=COMMENT1_CONTENT)
+    comment1 = Comment(author=users[0], post_parent=post0, comment_parent=comment0, content=COMMENT2_CONTENT)
+    return [comment0, comment1]
+
+
+@pytest.mark.django_db
+class TestCommentModel:
+    # Testing different comment attributes
+
+    def test_new_comment_properties(self, comments, users, post0):
```
Comment (Yarboa) [inline feed/tests/test_comment_model.py:38]: I assume the test check, comments created correct,
Why not filter for specific comments and compare it with fixtures return value?

File: feed/tests/test_comment_model.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        # Testing comment data
+        assert comments[0].author == users[1]
+        assert comments[0].post_parent == post0
+        assert comments[0].content == COMMENT1_CONTENT
+        assert comments[1].comment_parent == comments[0]
+
+    def test_comment_save_to_db(self, comments):
+        # Testing comment save in DB
+        comments[0].save()
+        assert comments[0] in Comment.comments.all()
```
Comment (Yarboa) [inline feed/tests/test_comment_model.py:48]: same here, the tests sgould filter the comment in db, and compare it against the returned data from fixtures

Please apply for all the rest of the tests

**Review by Yarboa — CHANGES_REQUESTED** [review]

**Comment by Yarboa** [comment]
@paOmer please take a look at previous comments

File: feed/tests/test_comment_model.py (hunk 1 above)
Comment (paOmer) [reply feed/tests/test_comment_model.py:38]: @Yarboa 
In this case if I'd like to filter the specific comment i'll need to save the fixture to DB first, did you mean't this?
and if so, by filtering the specific comment, you mean I should filter the comment like in my reply below?
Thanks ahead

Comment (Yarboa) [reply feed/tests/test_comment_model.py:38]: No, comments are the returned object by the Model creation
Now, the test search for the comment with filter help, or manager api, and compare it to the fixture data

File: feed/tests/test_comment_model.py (hunk 2 above)
Comment (Yarboa) [reply feed/tests/test_comment_model.py:48]: Yes, but please compare it with all comments data

File: feed/tests/test_post_model.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import django.contrib.auth
+
+
+User = django.contrib.auth.get_user_model()
+POST_TITLE = "Some First Post Title"
+POST_CONTENT = "Some first post content here"
+
+
+@pytest.fixture
+def users(db):
```
Comment (Yarboa) [inline feed/tests/test_post_model.py:12]: You can use random user creation here with random passwords
each time fixture is called you have user, and users are returned

File: feed/tests/test_post_model.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+@pytest.fixture
+def post0(db, users):
+    return Post.posts.create(title=POST_TITLE, content=POST_CONTENT, author=users[0])
+
+
+@pytest.mark.django_db
+class TestPostModel:
+    # Testing different post attributes
+
+    def test_new_post(self, users, post0):
```
Comment (Yarboa) [inline feed/tests/test_post_model.py:27]: The test checks user creation, 
just filter uses and assert for the created user against fixture data

same goes for post

File: feed/tests/test_post_model.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        # Testing post data
+        assert post0.title == POST_TITLE
+        assert post0.content == POST_CONTENT
+        assert post0.author == users[0]
+
+    def test_post_save(self, post0):
+        # Testing post save in DB
+        post0.save()
```
Comment (Yarboa) [inline feed/tests/test_post_model.py:35]: This should go into fixture of post
and then the tests just assert the fixture content against the post model filter,

File: feed/tests/test_post_model.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        assert post0 in Post.posts.main_feed()
+
+    def test_post_like(self, users, post0):
+        # Testing likes DB
+        post0.save()
+        post0.likes.add(users[1])
```
Comment (Yarboa) [inline feed/tests/test_post_model.py:41]: once you find the posts add the user returned

File: feed/tests/test_comment_model.py (hunk 1 above)
Comment (paOmer) [reply feed/tests/test_comment_model.py:42]: Done.

File: feed/tests/test_comment_model.py (hunk 2 above)
Comment (paOmer) [reply feed/tests/test_comment_model.py]: Im not sure what do you mean by compare to all comments data, do you mean to filter according to all properties and not only to content?

File: feed/tests/test_post_model.py (hunk 6 above)
Comment (paOmer) [reply feed/tests/test_post_model.py]: Done

File: feed/tests/test_post_model.py (hunk 5 above)
Comment (paOmer) [reply feed/tests/test_post_model.py]: Done

File: feed/tests/test_post_model.py (hunk 4 above)
Comment (paOmer) [reply feed/tests/test_post_model.py:29]: Done

File: feed/tests/test_post_model.py (hunk 3 above)
Comment (paOmer) [reply feed/tests/test_post_model.py:12]: Do you mean to do so by using additional package like this [link](https://eshlox.net/2019/06/11/create-random-data-for-your-django-unit-tests-using-model-mommy)?

Is it necessary?

**Comment by paOmer** [comment]
Hi @Yarboa, after you'll review I would squash and force push, as mentioned in the last Beyond session.

File: feed/tests/test_post_model.py (hunk 3 above)
Comment (Yarboa) [reply feed/tests/test_post_model.py:12]: consider this as an idea
```
import random
import string

print ( 'user_' + ''.join(random.choice(letters) for i in range(2)))
```

File: feed/tests/test_comment_model.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    return [comment0, comment1]
+
+
+@pytest.mark.django_db
+class TestCommentModel:
+    # Testing different comment attributes
+
+    def test_new_comment_properties(self, comments, users, post0):
+        # Testing comment data
+        comment0 = Comment.comments.filter(content=COMMENT1_CONTENT).first()
```
[the remark this answers is not in the document]
Comment (paOmer) [reply feed/tests/test_comment_model.py]: Done. Applied to all test files.

File: feed/tests/test_comment_model.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+    def test_comment_save_to_db(self, comments):
+        # Testing comment save in DB
+        comment = Comment.comments.filter(content=COMMENT1_CONTENT).first()
+        assert comment in Comment.comments.all()
+
+    def test_parent_comment_removal_to_child_comment(self, comments):
+        # Testing that removal of parent comment also removes child comment
+        # In this case, comments[0] is the parent of comment[1]
+        comment0 = Comment.comments.filter(content=COMMENT1_CONTENT).first()
```
[the remark this answers is not in the document]
Comment (paOmer) [reply feed/tests/test_comment_model.py]: Done and applied to all test files

File: feed/tests/test_post_model.py (hunk 3 above)
Comment (paOmer) [reply feed/tests/test_post_model.py:23]: Done and added to all test files, will move this and common fixtures to a single file in a future PR

File: feed/tests/test_post_comment_relation.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+from feed.models import Post, Comment
+import django.contrib.auth
+
+
+User = django.contrib.auth.get_user_model()
+POST_TITLE = "Some First Post Title"
+POST_CONTENT = "Some first post content here"
+
+
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (paOmer) [reply feed/tests/test_post_comment_relation.py:22]: Opened an issue #84

File: feed/tests/test_comment_model.py (hunk 2 above)
Comment (Yarboa) [reply feed/tests/test_comment_model.py]: This is the purpose of the test, 
Ig you have a comment, how do you find the logical connections from db, FYI

**Review by Yarboa — APPROVED** [review]
LGTM

**Pull request merged**